test/api/common/HttpClient.test.ts

```diff
@@ -0,0 +1,168 @@
+use(sinonChai);
+
+const fakeURL = 'http://foo.bar';
+const GET_ADDRESS_PATH = '/getaddr';
+const RELAY_PATH = '/relay';
+const POST_ESTIMATE = '/estimate';
+const VERIFIER_SUFFIX = '?verifier=';
+
+describe('HttpClient', function () {
+  afterEach(function () {
+    sandbox.restore();
+  });
+  describe('constructor', function () {
+    it('should store http wrapper', function () {
+     const expectedHttpWrapper = new HttpWrapper();
+      const httpClient = new HttpClient(expectedHttpWrapper);
+
+      expect((httpClient as unknown as {
+        _httpWrapper: HttpWrapper
+      })._httpWrapper).to.be.equal(expectedHttpWrapper);
+    });
+  });
+
+  describe('getChainInfo', function () {
+    it(`should call httpWrapper.sendPromise with given url + '${GET_ADDRESS_PATH}'`, async function () {
+      const expectedSendPromiseUrl = fakeURL + GET_ADDRESS_PATH;
+      const httpWrapperSendPromiseSpy = sandbox
+        .stub(HttpWrapper.prototype, 'sendPromise').resolves({ foo: 'bar' });
+      const httpClient = new HttpClient(new HttpWrapper());
+      await httpClient.getChainInfo(fakeURL);
+
+      expect(httpWrapperSendPromiseSpy).to.be.called;
+      expect(httpWrapperSendPromiseSpy).to.be.calledWith(
+        expectedSendPromiseUrl,
+      );
+    });
+
+    it(`should call httpWrapper.sendPromise with given url + '${GET_ADDRESS_PATH}' + given verifier`, async function () {
+      const verifier = '0x123';
+      const expectedSendPromiseUrl = fakeURL + GET_ADDRESS_PATH + VERIFIER_SUFFIX + verifier;
+      const httpWrapperSendPromiseSpy = sandbox
+        .stub(HttpWrapper.prototype, 'sendPromise').resolves({ foo: 'bar' });
+      const httpClient = new HttpClient(new HttpWrapper());
+      await httpClient.getChainInfo(fakeURL, verifier);
+
+      expect(httpWrapperSendPromiseSpy).to.be.called;
+      expect(httpWrapperSendPromiseSpy).to.be.calledWith(
+        expectedSendPromiseUrl,
+      );
+    });
+
+    it('should return response data', async function () {
+      const expectedResponse = { foo: 'bar' };
+      sandbox
+        .stub(HttpWrapper.prototype, 'sendPromise')
+        .resolves(expectedResponse);
+      const httpClient = new HttpClient(new HttpWrapper());
+
+      const response = await httpClient.getChainInfo(fakeURL);
+
+      expect(response).to.be.equal(expectedResponse);
+    });
+
+    it('should throw error if response contains message field', async function () {
+      sandbox
+        .stub(HttpWrapper.prototype, 'sendPromise')
+        .resolves({ message: 'bar' });
+      const httpClient = new HttpClient(new HttpWrapper());
+
+      await expect(httpClient.getChainInfo(fakeURL)).to.be.rejectedWith(
+        'Got error response from relay: bar',
+      );
+    });
+  });
+
+  describe('relayTransaction', function () {
+    it(`should call httpWrapper.sendPromise with given url + ${RELAY_PATH}`, async function () {
+      const request = { foo: 'bar' } as unknown as EnvelopingTypes.DeployRequestStruct;
```

Comment:
Should not be using the whole DeployRequestStruct, to make the unit test more easy to understand?

Reply:
IMHO not necessary, including additional properties may make me understand they're required for this test.